### PR TITLE
Use new Maven plugins, add Gradle support

### PR DESCRIPTION
**Pull request by wsh**

### CODE REVIEW

File: final/settings.gradle [hunk 1]
```diff
@@ -0,0 +1 @@
+rootProject.name = 'guestbook'
```
Comment (lesv) [inline final/settings.gradle]: Probably needs newline

File: phase1/build.gradle [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    promote = true              // default - & make this the current version
+  }
+}
+
+group = 'com.example.guestbook'   // Generated output GroupId
+version = '1.0-SNAPSHOT'         // Version in generated output
+
+sourceCompatibility = 1.7
+targetCompatibility = 1.7
+// [END gradle]
```
Comment (lesv) [inline phase1/build.gradle]: newline

File: phase1/settings.gradle [hunk 3]
```diff
@@ -0,0 +1 @@
+rootProject.name = 'guestbook'
```
Comment (lesv) [inline phase1/settings.gradle]: newline

File: README.md [hunk 4]
```diff
@@ -14,7 +14,7 @@ Please ask questions on [Stackoverflow](http://stackoverflow.com/questions/tagge
 How do I, as a developer, start working on the project?
 
 1. cd final  (or phase1)
-1. `mvn clean appengine:devserver`
+1. `mvn clean appengine:run` OR `./gradlew appengineRun`
```
Comment (lesv) [inline README.md]: I prefer `gradle appengineRun` and a mention of ./gradlew.sh for folks who don't have Gradle installed.  But I'm Ok w/ this if you insist.

File: final/build.gradle [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  jcenter()
+  mavenCentral()
+}
+
+apply plugin: 'java'                              // standard Java tasks 
+apply plugin: 'war'                               // standard Web Archive plugin
+apply plugin: 'com.google.cloud.tools.appengine'  // App Engine tasks
+
+dependencies {
+  providedCompile 'javax.servlet:javax.servlet-api:3.1.0'
```
Comment (lesv) [inline final/build.gradle]: 2.5 for standard, 3.1.0 for flex

File: final/build.gradle [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  testCompile 'com.google.appengine:appengine-testing:+'
+  testCompile 'com.google.appengine:appengine-api-stubs:+'
+}
+
+appengine {  // App Engine tasks configuration
+  run {      // local (dev_appserver) configuration (standard environments only)
+    port = 8080                 // default
+  }
+
+  deploy {   // deploy configuration
```
Comment (lesv) [inline final/build.gradle]: These should go, as they are the defaults, and actually impede things in some cases.

File: final/build.gradle [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    promote = true              // default - & make this the current version
+  }
+}
+
+group = 'com.example.guestbook'   // Generated output GroupId
+version = '1.0-SNAPSHOT'          // Version in generated output
+
+sourceCompatibility = 1.7
+targetCompatibility = 1.7
+// [END gradle]
```
Comment (lesv) [inline final/build.gradle]: newline

File: final/build.gradle [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+apply plugin: 'java'                              // standard Java tasks 
+apply plugin: 'war'                               // standard Web Archive plugin
+apply plugin: 'com.google.cloud.tools.appengine'  // App Engine tasks
+
+dependencies {
+  providedCompile 'javax.servlet:javax.servlet-api:3.1.0'
+  compile 'com.google.appengine:appengine-api-1.0-sdk:+'
+  compile group: 'javax.servlet.jsp.jstl', name: 'jstl-api', version: '1.2'
```
Comment (lesv) [inline final/build.gradle]: Objectify doesn't appear as a dependency.

File: phase1/build.gradle [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  jcenter()
+  mavenCentral()
+}
+
+apply plugin: 'java'                              // standard Java tasks 
+apply plugin: 'war'                               // standard Web Archive plugin
+apply plugin: 'com.google.cloud.tools.appengine'  // App Engine tasks
+
+dependencies {
+  providedCompile 'javax.servlet:javax.servlet-api:3.1.0'
```
Comment (lesv) [inline phase1/build.gradle]: 2.5 for standard

**Review by lesv — COMMENTED** [review]
PTAL - Thanks

File: phase1/build.gradle (hunk 9 above)
Comment (wsh) [inline phase1/build.gradle]: /facepalm thanks

File: final/build.gradle (hunk 7 above)
Comment (wsh) [inline final/build.gradle]: ack

File: final/build.gradle (hunk 6 above)
Comment (wsh) [inline final/build.gradle]: ack

File: final/build.gradle (hunk 8 above)
Comment (wsh) [inline final/build.gradle]: ack, sorry!

File: final/build.gradle (hunk 5 above)
Comment (wsh) [inline final/build.gradle]: ack

File: phase1/settings.gradle (hunk 3 above)
Comment (wsh) [inline phase1/settings.gradle]: ack

File: phase1/build.gradle (hunk 2 above)
Comment (wsh) [inline phase1/build.gradle]: ack

File: final/settings.gradle (hunk 1 above)
Comment (wsh) [inline final/settings.gradle]: ack

**Review by lesv — APPROVED** [review]
LGTM - we may not wish to merge just yet, as this is probably used in our docs somewhere.  But good to create the Veneer one from.

**Comment by wsh** [comment]
Okay, rad. What do we need to get out of the way to merge this?

**Pull request closed without merging**